pkg/operator/hive/hive_controller.go

```diff
@@ -302,10 +303,15 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		r.(*ReconcileHiveConfig).hiveImagePullPolicy = pullPolicy
 		nodeSelector := operatorDeployment.Spec.Template.Spec.NodeSelector
 		log.Debugf("loaded nodeSelector from hive-operator deployment: %v", nodeSelector)
-		r.(*ReconcileHiveConfig).nodeSelector = nodeSelector
 		tolerations := operatorDeployment.Spec.Template.Spec.Tolerations
 		log.Debugf("loaded tolerations from hive-operator deployment: %v", tolerations)
-		r.(*ReconcileHiveConfig).tolerations = tolerations
+		imagePullSecrets := operatorDeployment.Spec.Template.Spec.ImagePullSecrets
```

Comment:
Does this work? I wouldn't have expected the Deployment to grow imagePullSecrets from an associated ServiceAccount -- I would have thought that would only happen at the Pod level.

We could probably get rid of this Deployment `Get()` and reuse `ReadSharedConfigFromThisPod()` instead. (We would need to add pod ns/name env [injection](https://github.com/openshift/hive/blob/66a6c9f9dc9184818547b9cc16855e8b7ee1c525/config/controllers/deployment.yaml#L50-L57).)